src/Hopper/Internal/LoweredCore/EvalClosureConvertedANF.hs

```diff
@@ -174,25 +194,30 @@ localEnvLookup
 localEnvLookup env controlStack var@(LocalNamelessVar depth (BinderSlot slot))
   = go env depth
   where
-    go :: EnvStackCC -> Word32  -> EvalCC c s Ref
-    go EnvEmptyCC _ = trace "here0" $ throwEvalError (\n ->
+    go :: EnvStackCC -> Word32 -> EvalCC c s Ref
+    go EnvEmptyCC n = throwEvalError (\n ->
       BadVariableCC (LocalVar var) controlStack (InterpreterStepCC n))
-    go (EnvConsCC theRefVect _) 0 = trace "here1" $ maybe
+    go (EnvConsCC theRefVect _) 0 = maybe
       (throwEvalError (\n ->
         BadVariableCC (LocalVar var) controlStack (InterpreterStepCC n)))
-      (return . traceShowId)
-      (theRefVect V.!?  fromIntegral slot)
-    go (EnvConsCC _ rest) w = trace "here2" $ go rest (w - 1)
+      return
+      (theRefVect V.!? fromIntegral slot)
+    go (EnvConsCC _ rest) w = go rest (w - 1)
```

Comment:
to make this a bit easier to read i'd suggest something like the following 
```haskell 
  go (EnvConsCC theRefVect _) 0 
       | Just val <-theRefVect V.!? fromIntegral slot  = return val 
       | otherwise  =   throwEvalError 
              (\step  ->  BadVariableCC 
                  (LocalVar var) controlStack $  InterpreterStepCC step )	
```
or something along those lines, might read a bit simpler.
Theres a valid argument to use a case on the lookup instead too,
but generally if the error case code is too "big", the 'maybe' combinator gets wonky